packages/wrangler/src/deployment-bundle/esbuild-plugins/hybrid-nodejs-compat.ts

```diff
@@ -169,71 +169,61 @@ function handleNodeJSGlobals(
 	build: PluginBuild,
 	inject: Record<string, string | string[]>
 ) {
-	const UNENV_GLOBALS_RE = /_virtual_unenv_global_polyfill-([^.]+)\.js$/;
+	const UNENV_GLOBALS_RE = /_virtual_unenv_global_polyfill-(.+)$/;
 	const prefix = nodePath.resolve(
 		getBasePath(),
 		"_virtual_unenv_global_polyfill-"
 	);
 
+	/**
+	 * Map of module identifiers to
+	 * - `injectedName`: the name injected on `globalThis`
+	 * - `exportName`: the export name from the module
+	 * - `importName`: the imported name
+	 */
+	const injectsByModule = new Map<
+		string,
+		{ injectedName: string; exportName: string; importName: string }[]
+	>();
+
+	const moduleIdSpecifier = new Map<string, string>();
+
+	for (const [injectedName, moduleSpecifier] of Object.entries(inject)) {
+		const [module, exportName, importName] = Array.isArray(moduleSpecifier)
+			? [moduleSpecifier[0], moduleSpecifier[1], moduleSpecifier[1]]
+			: [moduleSpecifier, "default", "defaultExport"];
+
+		if (!injectsByModule.has(module)) {
+			injectsByModule.set(module, []);
+			moduleIdSpecifier.set(module.replaceAll("/", "-"), module);
+		}
+		// eslint-disable-next-line  @typescript-eslint/no-non-null-assertion
+		injectsByModule.get(module)!.push({ injectedName, exportName, importName });
+	}
+
 	build.initialOptions.inject = [
 		...(build.initialOptions.inject ?? []),
 		//convert unenv's inject keys to absolute specifiers of custom virtual modules that will be provided via a custom onLoad
-		...Object.keys(inject).map(
-			(globalName) => `${prefix}${encodeToLowerCase(globalName)}.js`
-		),
+		...[...moduleIdSpecifier.keys()].map((id) => `${prefix}${id}`),
 	];
 
 	build.onResolve({ filter: UNENV_GLOBALS_RE }, ({ path }) => ({ path }));
 
 	build.onLoad({ filter: UNENV_GLOBALS_RE }, ({ path }) => {
 		// eslint-disable-next-line  @typescript-eslint/no-non-null-assertion
-		const globalName = decodeFromLowerCase(path.match(UNENV_GLOBALS_RE)![1]);
-		const { importStatement, exportName } = getGlobalInject(inject[globalName]);
+		const module = moduleIdSpecifier.get(path.match(UNENV_GLOBALS_RE)![1])!;
+		// eslint-disable-next-line  @typescript-eslint/no-non-null-assertion
+		const injects = injectsByModule.get(module)!;
```

Comment:
In general I prefer to use runtime asserts rather than `!` assertions since it allows us to control the error shown to give better context:

```suggestion
		const injects = injectsByModule.get(module);
		assert(injects, `Expected ${module} to map to an inject`);
```

this also avoids the need to add eslint overrides.